api/authenticator.py

```diff
@@ -936,49 +971,87 @@ def __init__(self, library, integration, analytics=None):
                 regexp = None
         self.external_type_regular_expression = regexp
 
+        field = ConfigurationSetting.for_library_and_externalintegration(
+            _db, self.LIBRARY_IDENTIFIER_FIELD, library, integration
+        ).value
+        if isinstance(field, basestring):
+            field = field.strip()
+        self.library_identifier_field = field
+
+        self.library_identifier_restriction_type = ConfigurationSetting.for_library_and_externalintegration(
+            _db, self.LIBRARY_IDENTIFIER_RESTRICTION_TYPE, library, integration
+        ).value
+        if not self.library_identifier_restriction_type:
+            self.library_identifier_restriction_type = self.LIBRARY_IDENTIFIER_RESTRICTION_TYPE_NONE
+
         restriction = ConfigurationSetting.for_library_and_externalintegration(
-            _db, self.PATRON_IDENTIFIER_RESTRICTION, library, integration
+            _db, self.LIBRARY_IDENTIFIER_RESTRICTION, library, integration
         ).value
-        if restriction and restriction.startswith("^"):
-            # Interpret the value a regular expression
+
+        if self.library_identifier_restriction_type == self.LIBRARY_IDENTIFIER_RESTRICTION_TYPE_REGEX:
             try:
-                restriction = re.compile(restriction)
+                self.library_identifier_restriction = re.compile(restriction)
             except Exception, e:
-                self.log.error(
-                    "Could not interpret identifier restriction as a regular expression: %r", e
-                )
-        self.patron_identifier_restriction = restriction
+                self.log.error("Could not interpret patron record restriction as a regular expression: %r", e)
```

Comment:
I think this should maybe raise an Exception rather than logging the error - not sure if it would be worse to block actual library patrons or to let in patrons of other libraries when there's a typo in the regex.

And this isn't necessary for this branch, but: it might be worth adding a `type: regex` to settings so that the admin interface controller code could validate that the expression compiles before it's saved to the db.

Reply:
I added a check for the regex to the controller, so it won't save if the regex is invalid. I also added a test for it. Since there is this validation do you think we still need to throw an exception here?

Reply:
Nice! With validation this should never happen. But if you did somehow end up with a bad regex in the db I think it would still be better to throw an exception than log it.